source/code/plugins/in_npmd.rb

```diff
@@ -0,0 +1,676 @@
+class Logger
```

Comment:
This file seems too big... can you split the business logic from configuration and helpers?